src/experiment.py

```diff
@@ -1,226 +1,58 @@
-                transform=self.get_transforms(
-                    mode="valid",
-                    image_size=image_size
-                ),
-            )
-
-            datasets["train"] = {
-                "dataset": train_set,
-                "collate_fn": collate_fn
-            }
-            tqdm.write(f"\nTrain dataset len: {len(train_set)}")
-
-            datasets["valid"] = {
-                "dataset": valid_set,
-                "collate_fn": collate_fn
-            }
-            tqdm.write(f"\nValid dataset len: {len(valid_set)}")
-
-            if sampler_params is not None:
-                safitty.set(
-                    datasets, "train", "sampler",
-                    value=MiniEpochSampler(
-                        data_len=len(train_set), **sampler_params
-                    )
-                )
-        else:
-            collate_fn = FilteringCollateFn("bboxes", "labels", "filename", "orig_image")
-            infer_set = DetectionDataset(
-                num_classes,
-                down_ratio,
-                max_objs,
-                valid_filepaths + train_filepaths,
-                valid_bboxes + train_bboxes,
-                valid_labels + train_labels,
-                transform=self.get_transforms(
-                    mode="valid",
-                    image_size=image_size
-                ),
-            )
-            datasets["infer"] = {
-                "dataset": infer_set,
-                "collate_fn": collate_fn
-            }
-
-        return datasets
+        def process_kwargs_by_default_values(parameter, default_parameter):
+            if parameter not in kwargs:
+                if default_parameter not in kwargs:
+                    raise ValueError('You must specify \"{}\" or default value(\"{}\") in config'
+                                     .format(parameter, default_parameter))
+                else:
+                    kwargs[parameter] = kwargs[default_parameter]
+
+        process_kwargs_by_default_values('train_annotation_file', 'annotation_file')
+        process_kwargs_by_default_values('valid_annotation_file', 'annotaiton_file')
+        process_kwargs_by_default_values('train_images_dir', 'images_dir')
+        process_kwargs_by_default_values('valid_images_dir', 'images_dir')
+
+        train_dataset = DetectionDataset(annotation_file=kwargs['train_annotation_file'],
+                                         images_dir=kwargs['train_images_dir'],
+                                         down_ratio=kwargs['down_ratio'],
+                                         max_objects=kwargs['max_objs'],
+                                         num_categories=kwargs['num_categories'],
+                                         image_size=kwargs['image_size'],
+                                         transform=train_transform(kwargs['image_size'][0])
+                                         )
+
+        # TODO TRAIN IS NOW EQUAL TO VAL
```

Comment:
why? 😄

Reply:
Because current config file example shows: 
```
   # default values, will be used if something aren't specified
    annotation_file: ./data/annotation.json
    images_dir: ./data/annotation.json

    # You may specify next parameters, data source will be overwritten
    #train_annotation_file: ./data_train/annotation.json
    #valid_annotation_file: ./data_valid/annotation.json
    #train_images_dir: ./data_train/images/
    #valid_images_dir: ./data_valid/images/
```
I'am able to fix it here